src/toil/wdl/wdltoil.py

```diff
@@ -2297,12 +2331,44 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
 
         # Run the workflow. We rely in this to handle entering the input
         # namespace if needed, or handling free-floating inputs.
-        workflow_job = WDLWorkflowJob(self._workflow, [self._inputs], [self._workflow.name], self._namespace)
+        workflow_job = WDLWorkflowJob(self._workflow, [self._inputs], [self._workflow.name], self._namespace, self._execution_dir)
         workflow_job.then_namespace(self._namespace)
         self.addChild(workflow_job)
         self.defer_postprocessing(workflow_job)
         return workflow_job.rv()
 
+# Monkey patch miniwdl's WDL.Value.Base.coerce() function
+# miniwdl recognizes when a string needs to be converted into a file
+# However miniwdl's string to file conversions is to just store the filepath
+# Toil needs to virtualize the file into the jobstore
+# So monkey patch coerce to always virtualize whenever a file is expected
+#   _virtualize_filename should detect if the value is already a file and return immediately if so
+# Sometimes string coerce is called instead, so monkeypatch string coerce
```

Comment:
I think some of this should be a docstring, and some looks like it ought to be comments inside the actual function.